drivers/hmis/spec/requests/hmis/ce/ce_referral_spec.rb

```diff
@@ -98,13 +98,130 @@
 
       it 'shows second step as available' do
         _, result = post_graphql(**variables) { query }
-        expect(response.status).to eq 200
+        expect(response.status).to eq(200), result.inspect
         steps = result.dig('data', 'ceReferral', 'steps')
 
         expect(steps[0]['status']).to eq('completed')
         expect(steps[1]['status']).to eq('available')
       end
     end
+
+    context 'when workflow includes a conditional step' do
+      let(:gateway) do
+        create(
+          :hmis_workflow_definition_gateway,
+          template: workflow_template,
+          gateway_type: 'exclusive',
+          name: 'conditional gw',
+        )
+      end
+
+      before do
+        client_acceptance_task.outflows.destroy_all
+        client_acceptance_task.connect_to!(gateway)
+        gateway.connect_to!(provider_acceptance_task, condition: 'needs_provider_acceptance = 1')
+        gateway.connect_to!(accept_referral, condition: 'needs_provider_acceptance = 0')
+      end
+
+      it 'does not return the conditional step until it is enabled' do
```

Comment:
nit on test naming, since this case _doesn't_ test that the step  is returned when its enabled
```suggestion
      it 'does not return the conditional step' do
```